parser.c

```diff
@@ -339,13 +339,36 @@ static struct kanshi_profile_output *parse_profile_output(
 }
 
 static struct kanshi_profile *parse_profile(struct kanshi_parser *parser) {
+	struct kanshi_profile *profile = calloc(1, sizeof(*profile));
+	wl_list_init(&profile->outputs);
+
+	// First parse an optional profile name
+	parser->tok_str_len = 0;
+	if (!parser_read_str(parser)) {
+		fprintf(stderr, "expected new profile, got %s\n",
+			token_type_str(parser->tok_type));
+		return NULL;
+	}
+	profile->name = (parser->tok_str_len == 0)? NULL: strdup(parser->tok_str);
+
+	// Then parse the opening bracket
 	if (!parser_expect_token(parser, KANSHI_TOKEN_LBRACKET)) {
 		return NULL;
 	}
 
-	struct kanshi_profile *profile = calloc(1, sizeof(*profile));
-	wl_list_init(&profile->outputs);
+	// Use the bracket position to generate a default profile name
+	if (profile->name == NULL) {
+		char generated_name[100];
+		int ret = snprintf(generated_name, 100,
+				"<anonymous at line %d, col %d>", parser->line, parser->col);
+		if (ret >= 0 && ret < 100) {
+			profile->name = strdup(generated_name);
+		} else {
+			profile->name = "<anonymous>";
```

Comment:
This would need to be `strdup`'ed. But honestly, I just wouldn't care

Reply:
It is just useless as profiles are never freed. But I can certainly do that, just in case someone implements config reload and tries to free the right pointers.

Reply:
Yeah